spec/main.tex

```diff
@@ -1100,158 +1809,203 @@ \subsubsection{Non-Threshold ($n$-out-of-$n$) Presigning} \label{section:orig-pr
-
-        \item Compute $\delta = \sum_j \delta_j$, and do:
-        \begin{itemize}
-            \item Assert $\delta \cdot G = \sum_j \Delta_j$. 
-            \item Set $R = \delta^{-1} \cdot \Gamma$ and output $(R, k_i, \chi_i)$.
-        \end{itemize}
-    \end{enumerate}
+    \begin{algorithmic}[1]
+        \State Receive $(\delta_j, \Delta_j, \psi''_{i,j})$ from all parties
+        \State Assert $\verifyni{log*}^L(\Encode(\sid, j), R_i, (N_j, K_j, \Delta_j, \Gamma), \psi''_{i,j})$
+            for all $j \in [n] \setminus \{i\}$
+        \State Compute $\delta = \sum_j \delta_j$
+        \State Assert $\delta \cdot G = \sum_j \Delta_j$
+        \State Set $R = \delta^{-1} \cdot \Gamma$ and \Return $(R, k_i, \chi_i)$
+    \end{algorithmic}
 \end{description}
 
 \subsubsection{Threshold ($t$-out-of-$n$) Presinging}
 
 \begin{description}
 
-\item[\bf Input.] 
-  Size of signing set $t$,
-  identities of parties in signing set,
-  index $i\in [t]$,
-  index map\footnote{$S(i)$ is the index that $P_i$ had at key-generation time.} $S : [t] \to [n]$,
-  key share $K_{S(i)}$,
-  context separation string $\sid$,
-  security level $L$, curve $\E$ with generator $G$.
-
-  {\bf HD-wallets inputs:} additive $\text{shift} \in \Fq$ \footnote{Deriving additive shift is up to specific standard of HD-wallets, e.g. see \cite{bip32} or \cite{slip10} \label{deriving-shift-note}} ($\text{shift} = 0$ disables HD derivation)
-
-    \item[\bf Setup.] The key share $K_{S(i)}$ contains $\threshold$, number of key holders~$n$, secret share $x'_{S(i)}$, parties' public shares $\vec X' = (X'_j)_{j \in [n]}$, a map $I : [n] \to \Fq \setminus \{0\}$, Paillier secret key $\sk_{S(i)}$, parties' Paillier keys $\vec N' = (N'_j)_{j \in [n]}$, and parties' auxiliary information $\vec R' = (s_j, t_j, \hat N)_{j \in [n]}$.
-
-    \item[\bf Step 1.] Set $\sk_i = \sk_{S(i)}$ and 
-                       $\vec R = (R'_{S(j)})_{j \in [n]}$. Then:
-    \begin{itemize}
-        \item If shares are additive\footnote{In this case we have $t=n$.} shares of the private key, 
-            set $x_{i} = x'_{S(i)}$, $\vec X = (X'_{S(j)})_{j \in [n]}$.
-        
-        \item If shares are Shamir secret shares of the private key: 
+    \item[\bf Input.] \
+    \begin{itemize}[noitemsep,topsep=0pt]
+        \item size of signing set $t \ge 2$,
+        \item party index $i \in [t]$,
+        \item injective index map $S : [t] \to [n]$ that maps signers indicies at signing to their indicies at key generation, i.e. $S(i)$ is the index that $P_i$ had at key-generation time,
+        \item key share $K_{S(i)}$,
+        \item context separation string $\sid \in \Byte^*$,
+        \item additive shift $\var{shift} \in \Z_q$ for HD child derivation. $\var{shift} = 0$ disables HD derivation. Deriving additive shift is up to specific standard of HD-wallets, e.g. see \cite{bip32} or \cite{slip10} \label{deriving-shift-note}
```

Comment:
The phrasing about "is up to a specific standard..." is a bit confusing. I would expand to something like "The procedure for deriving this shift from a human representation of an HD-wallet is out of scope for this algorithm; for specific standards additive shift derivation see ...". And probably move it to a footnote.

Also, I don't see the `\label` having any effect in the pdf

Reply:
> a human representation of an HD-wallet

Not sure this phrase makes sense though, I imagine many humans would argue that a list of indexes is not human-representation at all

Reply:
I'd keep this:

> The procedure for deriving the shift is out of scope for this algorithm; for specific standards for shift derivation see ...

But it says basically the same thing to me as what we have now

Reply:
I like your option. I don't think it's the same thing, this one explicitly says "out of scope"